exotica_core/src/kinematic_tree.cpp

```diff
@@ -1008,7 +1083,47 @@ void KinematicTree::SetFloatingBaseLimitsPosXYZEulerZYX(
     }
     for (int i = 0; i < 6; ++i)
     {
+        // Set velocity and acceleration to +inf
         controlled_joints_[i].lock()->joint_limits = {lower[i], upper[i]};
+        controlled_joints_[i].lock()->velocity_limit = {inf};
+        controlled_joints_[i].lock()->acceleration_limit = {inf};
+    }
+
+    UpdateJointLimits();
+}
+
+void KinematicTree::SetPlanarBaseLimitsPosXYEulerZ(
+    const std::vector<double>& lower, const std::vector<double>& upper, const std::vector<double>& velocity, const std::vector<double>& acceleration)
+{
+    bool velocity_acceleration_to_inf = false;
+    if (controlled_base_type_ != BaseType::PLANAR)
+    {
+        ThrowPretty("This is not a planar joint!");
+    }
+    if (velocity.size() == 0 && acceleration.size() == 0)
+    {
+        velocity_acceleration_to_inf = true;
+    }
+    else if (lower.size() != 3 || upper.size() != 3 || velocity.size() != 3 || acceleration.size() != 3)
+    {
+        ThrowPretty("Wrong limit data size!");
+    }
+    else
+    {
+    }
+    for (int i = 0; i < 3; ++i)
+    {
+        controlled_joints_[i].lock()->joint_limits = {lower[i], upper[i]};
+        if (velocity_acceleration_to_inf == false)
+        {
+            controlled_joints_[i].lock()->velocity_limit = {velocity[i]};
+            controlled_joints_[i].lock()->acceleration_limit = {acceleration[i]};
+        }
+        else
+        {
+            controlled_joints_[i].lock()->velocity_limit = {inf};
+            controlled_joints_[i].lock()->acceleration_limit = {inf};
+        }
```

Comment:
```
controlled_joints_[i].lock()->velocity_limit = {velocity.size() != 0 ? velocity[i] : inf};
controlled_joints_[i].lock()->acceleration_limit = {acceleration.size() != 0 ? acceleration[i] : inf};
```

Reply:
This is super neat - didn't know you could evaluate like this in cpp. Thanks!